### PR TITLE
new pkg: add samba, cifs-utils & dependencies

### LINKED ISSUE #11041: [Wolfi Package Request]: cifs-utils
### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

Hello,

The package I am looking for is cifs-utils. A link to the alpine version of the package can be found [here](https://pkgs.alpinelinux.org/package/edge/main/x86/cifs-utils).

### which versions of the software should we include?

7.0-r2 (or latest is fine)

### Add some justification for why this specific package and versions are important.

We have an application that needs to mount nfs and cifs/smb shares. Currently Wolfi supports NFS, but I did not see cifs-utils, so I am forced to use Alpine for this application when Wolfi would be preferred.

Thank you.

### Please check all that apply

- [ ] This package has an un-restrictive license
- [X] The package/versions proposed are actively maintained upstream
- [ ] I am interested in adding this package to Wolfi OS myself
- [X] I am willing to help maintain this package

### PR DESCRIPTION
**Pull request by tuananh**
this PR add samba, cifs-utils and their dependencies.

Fixes: #11041 

Related: #6550 

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

**Comment by tuananh** [comment]
@developer-guy i took over your work in the other PR and continue adding samba here :)

**Comment by tuananh** [comment]
@ajayk is it possible for me to trigger the CI run myself?

**Comment by tuananh** [comment]
looks like yam is doing something funky 

```
samba.yaml has a diff from the expected formatting
--- want
+++ got
@@ -408,8 +408,8 @@
           mv "${{targets.destdir}}"/usr/lib/libsmbldap.so.* "${{targets.contextdir}}"/usr/lib
           mv "${{targets.destdir}}"/usr/lib/libtevent-util.so.* "${{targets.contextdir}}"/usr/lib
 
 # FIXME: in latest release 4.19.4, autoconf check is failing for python lib like py3-tdb so disable update for now
 update:
-  enabled
+  enabled: false
   # release-monitor:
-  #   identifier: 4[7](https://github.com/wolfi-dev/os/actions/runs/7601914151/job/20702666619?pr=11504#step:5:8)5[8](https://github.com/wolfi-dev/os/actions/runs/7601914151/job/20702666619?pr=11504#step:5:9): false
+  #   identifier: 4758
```

**Comment by tuananh** [comment]
hmm not sure why local build works for me but CI aint

**Comment by tuananh** [comment]
@ajayk what to do in this case since samba is having a few CVEs?

**Comment by ajayk** [comment]
> @ajayk what to do in this case since samba is having a few CVEs?

@tuananh we will review this week

**Comment by xnox** [comment]
this is an impressive set!

File: samba.yaml [hunk 1]
```diff
@@ -0,0 +1,417 @@
+package:
+  name: samba
+  version: 4.19.4
```
Comment (jamonation) [inline samba.yaml:3]:
```suggestion
  version: 4.19.5
```

I see upstream [cut a new tag](https://gitlab.com/samba-team/samba/-/tags/samba-4.19.5), and it reduces the CVE count quite a bit.

File: samba.yaml [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+      expected-sha256: 4026d93b866db198c8ca1685b0f5d52793f65c6e63cb364163af661fdff0968c
+      uri: https://download.samba.org/pub/samba/stable/samba-${{package.version}}.tar.gz
+
+  - uses: patch
+    with:
+      patches: tinfo.patch
+
+  - uses: autoconf/configure
+    with:
+      opts: |
```
Comment (jamonation) [inline samba.yaml:72]:
```suggestion
      opts: |
        LDFLAGS="-ltinfo" \
```

So this is an interesting one - is the `tinfo.patch` above needed if this LDFLAGS approach is used?

**Comment by cpanato** [comment]
Is this still relevant?

**Comment by jamonation** [comment]
> Is this still relevant?

Yes, let's get this updated, passing CI and merged in. Samba, CIFS, there so many useful things here!

**Comment by tuananh** [comment]
@jamonation I just fixed a few missing deps and it builds locally now.

image

**Comment by tuananh** [comment]
Let's see if the CI is green after this. And then I can start adding tests.

**Comment by tuananh** [comment]
Getting an error while building `py3-testtools`

```
2024/07/09 23:46:03 WARN   File "/usr/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in <module>
2024/07/09 23:46:03 WARN     main()
2024/07/09 23:46:03 WARN   File "/usr/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 357, in main
2024/07/09 23:46:03 WARN     json_out["return_val"] = hook(**hook_input["kwargs"])
2024/07/09 23:46:03 WARN                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024/07/09 23:46:03 WARN   File "/usr/lib/python3.12/site-packages/pyproject_hooks/_in_process/_in_process.py", line 271, in build_wheel
2024/07/09 23:46:03 WARN     return _build_backend().build_wheel(
2024/07/09 23:46:03 WARN            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024/07/09 23:46:03 WARN   File "/tmp/build-env-tgeaxab7/lib/python3.12/site-packages/hatchling/build.py", line 58, in build_wheel
2024/07/09 23:46:03 WARN     return os.path.basename(next(builder.build(directory=wheel_directory, versions=['standard'])))
2024/07/09 23:46:03 WARN                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024/07/09 23:46:03 WARN   File "/tmp/build-env-tgeaxab7/lib/python3.12/site-packages/hatchling/builders/plugin/interface.py", line 155, in build
2024/07/09 23:46:03 WARN     artifact = version_api[version](directory, **build_data)
2024/07/09 23:46:03 WARN                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024/07/09 23:46:03 WARN   File "/tmp/build-env-tgeaxab7/lib/python3.12/site-packages/hatchling/builders/wheel.py", line 476, in build_standard
2024/07/09 23:46:03 WARN     record = archive.add_file(included_file)
2024/07/09 23:46:03 WARN              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024/07/09 23:46:03 WARN   File "/tmp/build-env-tgeaxab7/lib/python3.12/site-packages/hatchling/builders/wheel.py", line 98, in add_file
2024/07/09 23:46:03 WARN     zip_info = zipfile.ZipInfo(relative_path, cast(TIME_TUPLE, self.time_tuple))
2024/07/09 23:46:03 WARN                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024/07/09 23:46:03 WARN   File "/usr/lib/python3.12/zipfile/__init__.py", line 398, in __init__
2024/07/09 23:46:03 WARN     raise ValueError('ZIP does not support timestamps before 1980')
2024/07/09 23:46:03 WARN ValueError: ZIP does not support timestamps before 1980
```

Notice this line when fetching the code

```
2024/07/09 23:45:54 INFO [git checkout] tag 2.7.2 is ac6919f5ead31ed331710dff921211e9c7440840
2024/07/09 23:45:54 INFO [git checkout] Setting commit date to Jan 1, 1980 (SOURCE_DATE_EPOCH found 0)
```

**Comment by jamonation** [comment]
Ah @tuananh, take a look at `py3-hatch.yaml`  here in Wolfi - it has a workaround of setting the source date to a timestamp far in the future for this very issue

**Comment by tuananh** [comment]
got this error , will check later today. maybe problem araise when i bump samba version.

```
2024-07-10T01:26:54.6814344Z 2024/07/10 01:26:54 WARN mv: can't rename '/home/build/melange-out/samba/usr/lib/samba/libsamba-net*samba4.so': No such file or directory
```

**Comment by tuananh** [comment]
Finally, it's green :D

**Review by jamonation — APPROVED** [review]
This looks great, thanks for all the work to get this over the line! The CVEs look like false positives, most/all have been patched or don't apply. I'll file advisory data for them once this is merged.

**Pull request merged**